File: demo/src/main/java/com/example/demo/entity/ChatMessage.java
package com.example.demo.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Setter
@Getter
@Entity
@Table(name = "chat_message")
public class ChatMessage {

    // Getters and Setters
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String sender; // "user" or "ai"

    @Lob
    private String message;

    private LocalDateTime timestamp;

    // Default constructor required by JPA
    public ChatMessage() {
        this.timestamp = LocalDateTime.now();
    }

    // Optional convenience constructor
    public ChatMessage(User user, String sender, String message) {
        this.user = user;
        this.sender = sender;
        this.message = message;
        this.timestamp = LocalDateTime.now();
    }

}

File: thera_frontend/lib/models/chat_message.dart
class ChatMessage {
  final int id;
  final String sender; // "user" or "assistant"
  final String message;
  final DateTime timestamp;
  final bool isFallback;

  ChatMessage({
    required this.id,
    required this.sender,
    required this.message,
    required this.timestamp,
    this.isFallback = false,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json['id'],
      sender: json['sender'],
      message: json['message'],
      timestamp: DateTime.parse(json['timestamp']),
      isFallback: json['isFallback'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {'sender': sender, 'message': message};
  }
}
